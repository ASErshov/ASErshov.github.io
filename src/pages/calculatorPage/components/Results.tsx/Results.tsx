import React from "react";
import { Box, Button, Typography } from "@mui/material";
import { Results as ResultsType } from "pages/calculatorPage/types";
import styles from "./Results.module.scss";

interface Props {
  results: ResultsType;
  reset: () => void;
}

const Results: React.FC<Props> = ({ results, reset }) => {
  return (
    <Box
      sx={{
        backgroundColor: "secondary.dark",
      }}
      className={styles.mainWrapper}
    >
      <div>
        <div className={styles.resultWrapper}>
          <div className={styles.titleWrapper}>
            <Typography variant="body2" component="span" sx={{ color: "common.white" }}>
              Tip Amount
            </Typography>
            <Typography variant="body2" component="span" sx={{ color: "primary.dark", fontSize: 14 }}>
              /person
            </Typography>
          </div>
          <Typography className={styles.label} variant="h2" component="span" sx={{ color: "primary.main" }}>
            {`$  ${results.tip}`}
          </Typography>
        </div>
        <div className={styles.resultWrapper}>
          <div className={styles.titleWrapper}>
            <Typography variant="body2" component="span" sx={{ color: "common.white" }}>
              Total
            </Typography>
            <Typography variant="body2" component="span" sx={{ color: "primary.dark", fontSize: 14 }}>
              / person
            </Typography>
          </div>
          <Typography className={styles.label} variant="h2" component="span" sx={{ color: "primary.main" }}>
            {`$ ${results.total}`}
          </Typography>
        </div>
      </div>
      <Button
        className={styles.resultButton}
        sx={{
          color: "secondary.dark",
          backgroundColor: "primary.main",
          "&:hover": {
            backgroundColor: "primary.light",
          },
          "&:disabled": {
            backgroundColor: "primary.dark",
            opacity: "0.5",
          },
          fontSize: 20,
        }}
        variant="contained"
        onClick={reset}
      >
        RESET
      </Button>
    </Box>
  );
};

export default Results;
