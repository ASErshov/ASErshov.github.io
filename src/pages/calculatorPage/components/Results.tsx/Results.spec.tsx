import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Results from "./Results";
import renderer from "react-test-renderer";

/* eslint-disable @typescript-eslint/explicit-function-return-type */

it("Matches DOM Snapshot", () => {
  const MOCKED_RESULTS = { tip: "0.00", total: "0.00" };
  const MOCKED_RESET = jest.fn();
  const domTree = renderer.create(<Results results={MOCKED_RESULTS} reset={MOCKED_RESET} />).toJSON();
  expect(domTree).toMatchSnapshot();
});

test("renders the Results", () => {
  const MOCKED_RESULTS = { tip: "0.00", total: "0.00" };
  const MOCKED_RESET = jest.fn();
  render(<Results results={MOCKED_RESULTS} reset={MOCKED_RESET} />);
  expect(screen.getByText("Tip Amount")).toBeInTheDocument();
  expect(screen.getByText("Total")).toBeInTheDocument();
  expect(screen.getByText("RESET")).toBeInTheDocument();
});

test("change functions test", () => {
  const MOCKED_RESULTS = { tip: "1.00", total: "1.00" };
  const MOCKED_RESET = jest.fn();
  render(<Results results={MOCKED_RESULTS} reset={MOCKED_RESET} />);
  expect(screen.getAllByText("$ 1.00").length).toBe(2);

  const resetButton = screen.getByText("RESET");
  resetButton && fireEvent.click(resetButton, {});
});
