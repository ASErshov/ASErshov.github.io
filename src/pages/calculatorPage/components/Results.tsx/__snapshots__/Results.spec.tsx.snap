// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches DOM Snapshot 1`] = `
<div
  className="mainWrapper MuiBox-root css-4ozt90"
>
  <div>
    <div
      className="resultWrapper"
    >
      <div
        className="titleWrapper"
      >
        <span
          className="MuiTypography-root MuiTypography-body2 css-1cl5n3e-MuiTypography-root"
        >
          Tip Amount
        </span>
        <span
          className="MuiTypography-root MuiTypography-body2 css-d9r3jl-MuiTypography-root"
        >
          /person
        </span>
      </div>
      <span
        className="MuiTypography-root MuiTypography-h2 label css-wf8t3v-MuiTypography-root"
      >
        $  0.00
      </span>
    </div>
    <div
      className="resultWrapper"
    >
      <div
        className="titleWrapper"
      >
        <span
          className="MuiTypography-root MuiTypography-body2 css-1cl5n3e-MuiTypography-root"
        >
          Total
        </span>
        <span
          className="MuiTypography-root MuiTypography-body2 css-d9r3jl-MuiTypography-root"
        >
          / person
        </span>
      </div>
      <span
        className="MuiTypography-root MuiTypography-h2 label css-wf8t3v-MuiTypography-root"
      >
        $ 0.00
      </span>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium resultButton css-14y9evz-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    RESET
  </button>
</div>
`;
